client/state/sites/reducer.js

```diff
@@ -34,20 +43,44 @@ export function items( state = {}, action ) {
 			return Object.assign( {}, state, {
 				[ site.ID ]: site
 			} );
-
+		case SITES_RECEIVE:
+			return keyBy( action.sites, 'ID' );
 		case DESERIALIZE:
 			if ( isValidStateWithSchema( state, sitesSchema ) ) {
 				return state;
 			}
-
 			return {};
 	}
 
 	return state;
 }
 
+/**
+ * Tracks sites fetching state
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function fetchingItems( state = {}, action ) {
+	switch ( action.type ) {
+		case SITES_REQUEST:
+		case SITES_REQUEST_FAILURE:
+		case SITES_REQUEST_SUCCESS:
+			return Object.assign( {}, state, {
+				all: action.type === SITES_REQUEST
```

Comment:
Might it be simpler to create separate reducers for fetching singular site vs. all sites?